admin/database/postgres/postgres.go

```diff
@@ -1121,7 +1121,7 @@ func (c *connection) FindMagicAuthToken(ctx context.Context, id string) (*databa
 	if err != nil {
 		return nil, parseErr("magic auth token", err)
 	}
-	return res.AsModel()
+	return c.magicAuthTokenFromDTO(res)
```

Comment:
Can we make fetching/decrypting the secret optional? E.g. with a bool. And not do it in `FindMagicAuthToken`, which is only used for checking a token during auth, and can just rely on the secret_hash?

Reply:
I can do that but wouldn't it be confusing if `FindMagicAuthToken` being used somewhere else and caller does not get secret even though its in the `MagicAuthToken` struct ? Should we create another struct ?

Reply:
Maybe to keep it simple, you could add `withSecret bool` as an arg to `FindMagicAuthToken`?